src/utils.ts

```diff
@@ -152,3 +152,16 @@ export const updateLanguage = (countryCode: string) => {
         ;(window as UCWindow).UC_UI?.updateLanguage?.(countryCode)
     }
 }
+
+/**
+ * Returns true if the Usercentrics dialog is currently open.
+ */
+export const isOpen = (): boolean => {
+    if (IS_BROWSER) {
+        const usercentricsRoot = document.getElementById('usercentrics-root')
+        const dialog = usercentricsRoot?.shadowRoot?.querySelector('[role="dialog"]')
+        return !!dialog
+    }
+
+    return false
+}
```

Comment:
This feels like a bit of a hack, but the `window.UC_UI` doesn't seem to provide a builtin way to detect the dialog state. Also, the `#usercentrics-root` isn't configurable so at least it's very clear that this code claims the dialog is open if a dialog element exists inside the Usercentrics root.